addons/website_slides/models/slide_channel.py

```diff
@@ -402,6 +405,7 @@ def toggle_active(self):
         to_activate = self.filtered(lambda channel: not channel.active)
         res = super(Channel, self).toggle_active()
         if to_archive:
+            self.write({'is_published': False})
```

Comment:
Should be `to_archive.write({'is_published': False})` IMO
And you could also filter on those that are published as a bonus.